ci-operator/step-registry/cucushift/installer/check/vsphere/customized-resource/cucushift-installer-check-vsphere-customized-resource-commands.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+source "${SHARED_DIR}/govc.sh"
+function get_node_names() {
+   local node_type=$1
+   if [ -n "${node_type}" ]; then
+      mapfile -t  node_lists < <(oc get node --no-headers | grep ${node_type} | awk '{print $1}')
+   else
+      mapfile -t  node_lists < <(oc get node --no-headers | awk '{print $1}')
+   fi
+  
+   echo ${node_lists[*]}
+}
+
+function check_master_nodes_number() {
+
+#get node name
+master_lists=$(get_node_names "master")
+IFS=" "
+read -r -a master_lists <<< "${master_lists}"
+#get etcd pods names
+infra_id=$(oc get -o jsonpath='{.status.infrastructureName}' infrastructure cluster)
+mapfile -t etcd_pods_names < <(oc get pod -n openshift-etcd | grep Running | grep etcd-${infra_id} | awk '{print $1}')
+
+  if [[ ${#master_lists[@]} != "${#etcd_pods_names[@]}" && ${#etcd_pods_names[@]} != "${CONTROL_PLANE_REPLICAS}" ]]; then
+      echo "ERROR: The node number and etcd pod number are not equal to ${CONTROL_PLANE_REPLICAS}"
+      return 1
+  fi
+  count=0
+  for node in "${master_lists[@]}"; do
+      etcd="etcd-${node}"
+      if [[ ${etcd_pods_names[*]} =~ ${etcd} ]]; then
+	  count=$(($count+1))
+      fi
+  done
+  if [[ "${count}" != "${CONTROL_PLANE_REPLICAS}" ]] ; then
+     echo "ERROR: The number of masters in Ready status is ${#master_lists[@]}, etcd pod number is ${#etcd_pods_names[@]} should be ${CONTROL_PLANE_REPLICAS}" 
+     return 1
+  else 
+     return 0
+  fi  
+}
+
+function check_worker_nodes_number() {
+worker_lists=$(get_node_names "worker")
+IFS=" "
+read -r -a worker_lists <<< "${worker_lists}"
+if [[ ${#worker_lists[@]} != "${COMPUTE_NODE_REPLICAS}" ]]; then
+     echo "ERROR:The number of workers in Ready status is ${#worker_lists[@]}, should be ${COMPUTE_NODE_REPLICAS}"
+     return 1
+else
+     return 0
+fi
+
+}
+function check_node_resource() {
+    local node_type=$1 expected_disk_size=$2 expected_cpu_number=$3 expected_memory_size=$4 status=0
+    tmp=$(get_node_names $1)
```

Comment:
```suggestion
    node_lists=($(get_node_names $1))
```
We can easily get node list array, does not need the following two lines.